doc/whats_new/v0.22.rst

```diff
@@ -14,17 +14,53 @@ refer to
 :ref:`sphx_glr_auto_examples_release_highlights_plot_release_highlights_0_22_0.py`.
 
 
+Cleaning of the public API
+--------------------------
+
+Scikit-learn has a public API, and a private API.
+
+Any change to the public API is subject to a deprecation cycle of two minor
+versions. The private API isn't publicly documented and isn't subject to any
+deprecation cycle, so users should not rely on it.
+
+Whether a tool is private or public depends on whether you can import it
+without a leading underscore in the import path. For example
+``sklearn.pipeline.make_pipeline`` is public, while
+`sklearn.pipeline._name_estimators` is private.
+``sklearn.ensemble._gb.BaseEnsemble`` is private too because the whole `_gb`
+module is private.
```

Comment:
It'd be nice to also mention the classes/functions which are private because their name has a leading underscore, eventhough the path may be _public_.

Reply:
There is ``sklearn.pipeline._name_estimators``, were you thinking of something else?

Reply:
some `_BaseBlahBlah` I guess. I think most of them are gonna be in the `_filename` locations and don't matter anyway. But not for the ones in the root directory.